README.md

```diff
@@ -0,0 +1,71 @@
+Generate code from CDDL description
+===================================
+
+CDDL is a human-readable description language defined in an IETF RFC (https://tools.ietf.org/html/rfc8610).
+Using the CMake function `cddl_source()`, you can autogenerate C code that validates and decodes CBOR encoded instances of the structure described in a given .cddl file.
+
+The code generation is performed by the Python script cddl_gen.py in scripts/.
+The generated code depends on a simple CBOR decoding library (cbor_decode) in src/.
+There are tests for the code generation in tests/.
+The tests require Zephyr (If your shell is set up to build Zephyr samples, the tests should also build)
+
+Features
+========
+
+The generated code consists of:
+ - A header file containing typedefs corresponding to the types described in the CDDL, as well as functions declarations for decoding functions requested to be exposed (ENTRY_TYPES).
+ - A C file containing all the decoding code.
+   The decoding code is divided into functions (internal unless the type is an ENTRY_TYPE), and each function generally contains a single massive `if` statement which strings together calls into the cbor_decode library.
```

Comment:
Not a fan of "single massive". Explain it in words instead, or drop it from t he explanation